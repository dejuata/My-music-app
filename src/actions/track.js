import { types } from "../types/types";

export const setTracks = (tracks) => ({
    type: types.setTracks,
    payload: tracks
})

export const addTrackFavorite = (id) => ({
    type: types.addTrackFavorite,
    payload: id
})

export const removeTrackFavorite = (id) => ({
    type: types.removeTrackFavorite,
    payload: id
})

export const logoutTrack = () => ({
    type: types.logoutTrack
})