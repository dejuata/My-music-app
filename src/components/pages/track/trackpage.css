.track-detail {
    width: 100%;
    /* background-color: var(--color-white); */
    /* color: var(--color-dark); */
    border-radius: 12px;
    padding: 12px;
    box-sizing: border-box;
}

.track-detail__image {
    margin: 0;
    /* padding: 20px; */
    /* box-sizing: border-box; */
}

.track-detail__image img{
    width: 100%;
    object-fit: cover;
    border-radius: 12px;
}

.track-detail__info {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: space-between;
    margin-top: 24px;
}

.track-detail__info h2 {
    margin: 0 ;
    font-size: var(--xlg);
    font-weight: bold;
}

.track-detail__info p {
    margin: 0;
}

.track-detail__info p:nth-child(2) {
    margin-bottom: 32px;
    font-size: var(--lg);

}

.track-detail__info p:nth-child(3) {
    font-size: var(--md);
    text-align: justify;
    font-weight: 300;
    margin-bottom: 12px;

}

.track-detail__info p:nth-child(4) {
    margin-bottom: 32px;
    font-weight: 300;
    font-size: var(--sm);
}

.secondary-button {
    background-color: var(--color-white);
    border-radius: 8px;
    border: 1px solid var(--color-text);
    color: var(--color-green);
    cursor: pointer;
    font-size: var(--lg);
    font-weight: 500;
    text-align: center;
    padding: 20px;
    box-sizing: border-box;
    width: 100%;
    text-decoration: none;
}