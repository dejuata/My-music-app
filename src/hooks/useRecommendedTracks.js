// import React, { useContext, useEffect, useState } from "react"
// import { getRecommendedTracks } from "../api/selectors/getRecommendedTracks";
// import { AuthContext } from "../auth/authContext";

// export const useRecommendedTracks = ( category ) => {
//     const { user } = useContext(AuthContext);
//     const { token } = user;

//     const [state, setState] = useState({
//         data: [],
//         loading: true
//     })

//     useEffect(() => {
//         getRecommendedTracks(category, token)
//             .then( tracks => {
//                 setState({
//                     data: tracks,
//                     loading: false
//                 })
//             })
//     }, [category, token])

//     return state;
// }