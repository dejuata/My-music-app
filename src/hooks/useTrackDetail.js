import { useEffect, useRef, useState } from "react"
import { getTrackDetail } from "../api/requests/getTrackDetail";

export const useTrackDetail = ( trackId ) => {

    const isMounted = useRef(true);

    const [state, setState] = useState({
        data: {},
        loading: true
    })

    useEffect( () => {
        return () => {
            isMounted.current = false;
        }
    }, [])

    useEffect(() => {

        setState({ data: {}, loading: true });

        getTrackDetail(trackId)
            .then( tracks => {
                if ( isMounted.current ) {
                    setState({
                        data: tracks,
                        loading: false
                    })
                }
            })
    }, [trackId])

    return state;
}