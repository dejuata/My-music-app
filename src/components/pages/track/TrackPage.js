import React from 'react'
import ReactLoading from "react-loading";
import { useParams } from 'react-router-dom';
import { useTrackDetail } from '../../../hooks/useTrackDetail';
import { TrackDetail } from '../../organisms/TrackDetail/TrackDetail';


export const TrackPage = () => {

    const { trackId } = useParams();

    const { data:track, loading } = useTrackDetail(trackId);


    return (
        <>
            {

                loading
                ?
                    <div className='home__loading'>
                        <ReactLoading type={ 'bubbles' } color="#fff" />
                    </div>
                :
                    <TrackDetail {...track}/>

            }
        </>
    )
}
