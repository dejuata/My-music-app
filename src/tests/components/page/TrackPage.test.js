import { TrackPage } from "../../../components/pages/track/TrackPage";
import { mount } from 'enzyme';
import { useTrackDetail } from "../../../hooks/useTrackDetail";

jest.mock("../../../hooks/useTrackDetail")

describe('Test TrackPage', () => {

    test('Should be render TrackPage', () => {

        useTrackDetail.mockReturnValue({
            data: {},
            loading: true
        })

        const wrapper = mount(<TrackPage />)

        expect(wrapper).toMatchSnapshot();

    })

    test('Should be render info', () => {

        useTrackDetail.mockReturnValue({
            data: {
                id: '1',
                name: 'Desnúdate Mujer',
                name_short: 'Desnúdate Mujer',
                album: "Voy Pa'Encima",
                artists: 'Frankie Ruiz',
                duration: '4:47',
                image: 'https://i.scdn.co/image/ab67616d00001e021386b29ab009dd27c398d141',
                url: 'https//url-track'
            },
            loading: false
        })

        const wrapper = mount(<TrackPage />)

        expect( wrapper.find('.home__loading').exists() ).toBe( false );
        expect( wrapper.find('.o-track-detail').exists() ).toBe( true );
        expect( wrapper.find('.o-track-detail__body-title h2').text().trim() ).toBe( 'Desnúdate Mujer')

    })

})