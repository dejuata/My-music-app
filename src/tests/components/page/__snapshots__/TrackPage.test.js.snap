// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TrackPage Should be render TrackPage 1`] = `
<div
  className="home__loading"
>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 32 32\\">
  <circle transform=\\"translate(8 0)\\" cx=\\"0\\" cy=\\"16\\" r=\\"0\\"> 
    <animate attributeName=\\"r\\" values=\\"0; 4; 0; 0\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0\\"
      keytimes=\\"0;0.2;0.7;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.6 0.4 0.8\\" calcMode=\\"spline\\" />
  </circle>
  <circle transform=\\"translate(16 0)\\" cx=\\"0\\" cy=\\"16\\" r=\\"0\\"> 
    <animate attributeName=\\"r\\" values=\\"0; 4; 0; 0\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0.3\\"
      keytimes=\\"0;0.2;0.7;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.6 0.4 0.8\\" calcMode=\\"spline\\" />
  </circle>
  <circle transform=\\"translate(24 0)\\" cx=\\"0\\" cy=\\"16\\" r=\\"0\\"> 
    <animate attributeName=\\"r\\" values=\\"0; 4; 0; 0\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0.6\\"
      keytimes=\\"0;0.2;0.7;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.6 0.4 0.8\\" calcMode=\\"spline\\" />
  </circle>
</svg>
",
      }
    }
    style={
      Object {
        "fill": "#fff",
        "height": 64,
        "width": 64,
      }
    }
  />
</div>
`;
